### PR TITLE
Mysql integration

### PR DESCRIPTION
**Pull request by romesh-jaya**
.prettierrc.js was also added to format quotes as single quotes within js code (which is the standard).

### CODE REVIEW

File: src/components/swap/stockGraph.vue [hunk 1]
```diff
@@ -1,142 +1,234 @@
...
+        const response = await axios.get(
+          `${process.env.BACKEND_ENDPOINT}/?action=getCurrentTime`
+        );
+        const serverTime = response.data && response.data[0];
+        if (serverTime) {
+          debugModeOn && console.log('getServerTime', serverTime.CURRENTTIME);
+          this.currentServerTime = new Date(serverTime.CURRENTTIME);
         }
+      } catch {
+        this.$errorNotification('Error connecting to Database');
```
Comment (JPBeukes) [inline src/components/swap/stockGraph.vue:171]: I think we can make this a silent catch without an error notification. Maybe just do a `console.error`

**Review by JPBeukes — CHANGES_REQUESTED** [review]
Looks good! I made one small change request and I would still like to see it in action before I approve. (I weren't able to connect to the backend endpoint)

File: src/utils/swap.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
+/* Logic of processData()
+/* ==============================================
+/* Note: all times are relative to DB's current timestamp. The data fetched from the DB contains a column 'representative_time'.
+/*      This column contains the timestamp that represents the price values of that particular record.
+/*      It is the start of the hour in the case of 'daily', and start of the day in case of 'weekly' or 'monthly' time series.
+/*      In order words, the record fetched will contain the closing values for that particular hour or day.
+/*
+/* 1. We have to calculate the start and end date of the data for the graph.
+/*  End date - for daily, is the (start of the current hour - 1 hour). For weekly/monthly, it is the (start of the current day - 1 day)
+/*  Start date - for daily, it is (End date - 1 day). For weekly, it is (End date - 7 days). For monthly, it is (End date - 31 days).
```
Comment (walterheymans) [inline src/utils/swap.js:10]: Can you please align the indentation of the text on all lines after the first to the same level as the line with the number. For example:
```
 /* 1. We have to calculate the xxx
 /*    End date - for daily, is xxx
 /*    Start date - for daily, xxx
```
Visually it just looks better. Please also do that for the rest of the list.

File: src/utils/swap.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+/* 4. If there was no record for the start date, then we need to look at the previous values (for the duration of the constant
+/*  'goBackNumberOfDaysExtrapolation'), to see if there are any values to be found. If not, then use zero as the starting value for step 3.
+/*
+/* Underlying assumption:
+/* ======================
+/* The DB will only have records for either (token1 = EOS and token2 = TLOS) or (token1 = TLOS and token2 = EOS), but not a mixture of both types.
+*/
+
+import moment from 'moment';
+const goBackNumberOfDaysExtrapolation = 2;
```
Comment (walterheymans) [inline src/utils/swap.js:27]: What is the drawback of making this more? How much longer will it take if we make this, lets say 5?

**Review by walterheymans — COMMENTED** [review]
The implementation looks really good. Thanks for the hard work. I made some comments inline for you to look at. Some are just questions to help me understand some things better. Thanks for also making some of the style changes to make things more consistent.

**Comment by walterheymans** [comment]
@romesh-jaya another thing I just noticed. I see there are many commits that have exactly the same commit message. Should this actually have been a single commit? Normally I would also suggest explaining a little bit more about what was done in the commit, but since this repo does not really follow that, I think it is okay.

Can you please have a look at the commit history and try to squash some of the commits that have a similar purpose into a single commit. If they do not belong in the same commit, can you please just reword the commit to make sure they explain the changes differently.

If you do not know how to rebase, please just let me know. I would explain it to you clearly. Basically the idea is no not make a new commit if you need more changes in a previous commit. Instead you can go "back in time" and edit the commit with the new updates. You can do an interactive rebase using `git rebase -i HEAD~5` (where 5 is the number of commits you want to go back to). Then you can change the working from `pick` to `edit` or some other things depending on what you want to do. You can probable also Google it if you are unsure. Maybe just make a backup of you branch before you make a mistake.

**Comment by romesh-jaya** [comment]
> 
@walterheymans , In my previous jobs, they had configured github to squash all the commits when merging a feature branch. Please see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits. 

In my opinion, I think all my changes should go as a single commit when merging, since all of my commits together represent a single feature - which is mysql integration. Let me know if you aren't in agreement with this.

File: src/utils/swap.js (hunk 3 above)
Comment (romesh-jaya) [reply src/utils/swap.js:27]: The drawback of increasing 'goBackNumberOfDaysExtrapolation' will be that more data is returned from each SQL query. And more data will be processed in the frontend. This is of special significance in the 'daily' time series, where there will be upto 72 additional records returned in each query if we increase the value from 2 to 5. 

If this value is changed, then the backend PHP code also needs changing.

File: src/components/swap/stockGraph.vue (hunk 1 above)
Comment (romesh-jaya) [reply src/components/swap/stockGraph.vue]: Fixed

File: src/components/swap/stockGraph.vue [hunk 4]
```diff
@@ -1,142 +1,234 @@
...
+    async fetchAndProcessGraphData() {
+      if (!this.currentServerTime) {
+        return;
+      }
+      try {
+        const token0 = this.getPool.reserve0;
+        const token1 = this.getPool.reserve1;
+        const currentDate = this.currentServerTime.toISOString().split('T')[0];
+        const response = await axios.get(
+          `${process.env.BACKEND_ENDPOINT}/?action=getData&token1=${this.fromTokenSymbol}&token2=${this.toTokenSymbol}&timespan=${this.timeSeries}&currentDate=${currentDate}`
```
[the remark this answers is not in the document]
Comment (romesh-jaya) [reply src/components/swap/stockGraph.vue]: Fixed

File: src/components/swap/stockGraph.vue [hunk 5]
```diff
@@ -1,142 +1,234 @@
...
+              data[1].toPrecision(
+                token0.symbol === this.fromTokenSymbol
+                  ? token0.precision
+                  : token1.precision
+              )
+            ),
+          ])
+        );
+      } catch {
+        this.$errorNotification('Error processing graph data');
```
[the remark this answers is not in the document]
Comment (romesh-jaya) [reply src/components/swap/stockGraph.vue]: Fixed

File: src/utils/swap.js [hunk 6]
```diff
@@ -0,0 +1,227 @@
+/* Logic of processData()
+/* ==============================================
+/* Note: all times are relative to DB's current timestamp. The data fetched from the DB contains a column 'representative_time'.
```
[the remark this answers is not in the document]
Comment (romesh-jaya) [reply src/utils/swap.js]: Fixed

File: src/utils/swap.js (hunk 2 above)
Comment (romesh-jaya) [reply src/utils/swap.js]: Fixed

**Comment by walterheymans** [comment]
> @walterheymans , In my previous jobs, they had configured github to squash all the commits when merging a feature branch. Please see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits.
> 
> In my opinion, I think all my changes should go as a single commit when merging, since all of my commits together represent a single feature - which is mysql integration. Let me know if you aren't in agreement with this.

This makes sense. However, I think two commits make more sense, since not all changes are part of the new feature. Many of it is just style changes. Consider splitting it in two commits, with that I am happy.

When you do squash them, can you please just make the commit message clear about the new feature.

**Comment by romesh-jaya** [comment]
> > @walterheymans , In my previous jobs, they had configured github to squash all the commits when merging a feature branch. Please see https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits.
> > In my opinion, I think all my changes should go as a single commit when merging, since all of my commits together represent a single feature - which is mysql integration. Let me know if you aren't in agreement with this.
> 
> This makes sense. However, I think two commits make more sense, since not all changes are part of the new feature. Many of it is just style changes. Consider splitting it in two commits, with that I am happy.
> 
> When you do squash them, can you please just make the commit message clear about the new feature.

Ok sure. Let me close this PR and create a new one, since i have addressed all the outstanding issues except the commit messages.

**Pull request closed without merging**